tests/src/behaviors/tx-per-second.ts

```diff
@@ -349,18 +360,112 @@ const signalTrial = async (period, activeAgents: Agents, allPlayers: Player[], c
     return finishTime - start
 }
 
+const sendOnInterval = async (agents: Agents, channel, period: number, sendInterval: number) : Promise<number> => {
+    let totalSent = 0
+    const start = Date.now()
+    do {
+        const intervalStart = Date.now()
+        let messagePromises = new Array(agents.length)
+        for (let i = 0; i < agents.length; i++) {
+            messagePromises[i] = send(i, agents[i].cell, channel, "signal" )
+        }
+        await Promise.all(messagePromises)
+        const intervalDuration = (Date.now() - intervalStart)
+        console.log(`Took: ${intervalDuration}ms to send ${agents.length} messages`)
+        totalSent += agents.length
+        if (intervalDuration < sendInterval) {
+            console.log(`Waiting ${(sendInterval - intervalDuration}ms before sending again`)
+            await delay(sendInterval - intervalDuration)
+        }
+    } while((Date.now() - start) < period)
+    return totalSent
+}
+
+const phaseTrial = async (period: number, sendInterval: number, activeAgents: Agents, allPlayers: Player[], channel) => {
+    let totalAgents = activeAgents.length
```

Comment:
I find this variable name misleading because it could mean total number of agents in the network (including inactive)